content/articles/nodejs-network-requests-using-axios/index.md

```diff
@@ -0,0 +1,158 @@
+```
+
+With that done, let's go ahead and create our server.
+
+Create an index.js file and write the following code in it.
+
+```Javascript
+const express = require("express");
+const app = express();
+const axios = require("axios");
+
+app.listen(2400, () => {
+	console.log("Server started at port 2400");
+});
+```
+
+Run the server using this command.
+
+```bash
+node index
+```
+
+Now that our server is up and running, let's see what network requests are and what axios allows us to do.
+
+### Getting started with axios
+
+We have mentioned network requests but we haven't said what it is. As the name suggests, network requests are requests sent from a client to a server over the internet. The client in this case can also be another server. Axios allows us to make `HTTP` requests from both the browser and Node.js applications. It allows us to make both `GET` and `POST` requests which are the most used `HTTP` methods.
+
+Let's see how we can make these types of requests.
+
+### Get request
+
+All axios requests are created using the axios object we imported in the first code snippet. The axios object takes an object parameter. The object contains the request url and the http method associated with the request.
+
+Add the following code just before `app.listen` method.
+
+```Javascript
+app.get("/async", async (req, res) => {
+	try {
+		const response = await axios({
+			url: "https://dog.ceo/api/breeds/list/all",
+			method: "get",
+		});
+		res.status(200).json(response.data);
+	} catch (err) {
+		res.status(500).json({ message: err });
+	}
+});
+```
+
+In the above code, we have made an endpoint that makes a `GET` request to the dog api. The endpoint then returns the response it receives from the server. The axios object returns a promise but we are able to use the async/await method which makes our code appear sequential. However, you can still use the promise library. This is how the code will look like.
+
+```Javascript
+app.get("/promise", (req, res) => {
+	axios({
+		url: "https://dog.ceo/api/breeds/list/all",
+		method: "get",
+	})
+		.then(response => {
+			res.status(200).json(response.data);
+		})
+		.catch((err) => {
+			res.status(500).json({ message: err });
+		});
+});
+```
+
+Test the two endpoints using postman and observe the output.
+
+Alternatively, instead of writing the url and the method, you can call the method directly from the object and just pass in the url.
+
+Example;
+
+```Javascript
+axios.get("https://dog.ceo/api/breeds/list/all")
+```
+
+### Post requests
+
+For post requests, axios object takes in the url, method and the body to post. Since the dogs api does not allow us to make post requests, we are just going to have a look at how this request can be made but not with a real API.
+
+To make a post request, all you need to do is to call the post method from the axios object and pass in the body to be posted.
+
+Below is a code example using the async/await method.
+
+```Javascript
```

Comment:
You could use the [Json placeholder API](https://jsonplaceholder.typicode.com/) which has all HTTP methods instead of having multiple network requests to different servers. This shows some uniformity in the application and much better flow

Reply:
Thanks Linus.